addons/sale_loyalty/models/sale_order.py

```diff
@@ -928,7 +928,7 @@ def __try_apply_program(self, program, coupon, status):
                 all_points = [p for p in all_points if p]
                 partner = False
                 # Loyalty programs and ewallets are nominative
-                if program.is_nominative:
+                if program.is_nominative or (program.applies_on == 'future' and program.program_type == "loyalty"):
```

Comment:
Where is it coming from? I don't see that part in the spec.
Could you also add description in the comment above?

Reply:
According to the 3rd point added in the testing IN stage, the partner should be linked for a newly created loyalty card for the Use point on "Future Order". This should be done only for loyalty cards. Hence this part of the code was added.

If you think it is confusing, we can edit the description for that comment. What do you think?

Reply:
Ok, that is exactly what the condition you add does.
But, if I read the comment, loyalty program should already be taken into account by the existing condition, so I wonder if the fix shouldn't be to change the `_compute_is_nominative` function, to take this case as well. What do you think? Wouldn't it make more sense?